### PR TITLE
address Issue #19

### PR DESCRIPTION
**Pull request by sachseka**
Thanks for reporting this @schumaki  Hope the changes fix this issue?

### CODE REVIEW

**Comment by sachseka** [comment]
> Thanks for reporting this @schumaki Hope the changes fix this issue?

PS: one review should be sufficient @schumaki @franikowsp whoever is first... ;)

**Comment by schumaki** [comment]
Unfortunally, the issue is still persisting...

I get as a message 
── Check: Variables in Info 
ℹ The following 1 variable is not in dataset but in info (`subunit` in blocks). It will be ignored during check: `N16H11104a`

but still: 
Fehler in `[.data.frame`(dat, match(cases, dat[, id]), match(subunit, names(dat))) : 
  nicht definierte Spalten gewählt
  
I can run additional tests on monday to further pin the issue down.

**Comment by sachseka** [comment]
> Unfortunally, the issue is still persisting...
> 
> I get as a message ── Check: Variables in Info ℹ The following 1 variable is not in dataset but in info (`subunit` in blocks). It will be ignored during check: `N16H11104a`
> 
> but still: Fehler in `[.data.frame`(dat, match(cases, dat[, id]), match(subunit, names(dat))) : nicht definierte Spalten gewählt
> 
> I can run additional tests on monday to further pin the issue down.

Hmm, in my ue case there is no error enymore
grafik
Maybe there is another problem? Could you provide the path to you syntax?

**Comment by schumaki** [comment]
A MWE can be found here: T:\Kristoph\MWE check_design

**Comment by sachseka** [comment]
> A MWE can be found here: T:\Kristoph\MWE check_design

Thank you! It's intriguing... I recognize the issue, but I can't reproduce it with my example data at the moment. I'll have to investigate this further...

**Comment by schumaki** [comment]
Ah I see. I had expected the fix to exclude the item in all blocks, not just in one.

The issue is now fixed. A code review is still outstanding (by me or @franikowsp ) before merging the pull request.

File: R/checkDesign.r
Comment (schumaki) [inline R/checkDesign.r:1]: Looks fine for me. What do you think, @franikowsp ?

File: tests/testthat/test-checkDesign.R
Comment (schumaki) [inline tests/testthat/test-checkDesign.R:1]: I haven't much experience with test_that. 
Does this looks ok to you, @franikowsp ?

File: R/checkDesign.r
Comment (franikowsp) [reply R/checkDesign.r:1]: LGTM, too!

File: R/checkDesign.r [hunk 1]
```diff
@@ -129,7 +129,23 @@ checkDesign <- function(dat, booklets, blocks, rotation, sysMis = "NA", id = "ID
                      {.envvar {toOmit}}",
                      wrap = TRUE)
     }
-    dat <- dat[, -match(toOmit, names(dat))]
+    dat <- dat[,-c(which( names(dat) %in% toOmit))]
+  }
+
+  # Check for variables in the info (`blocks$subunit`) that are not in `dat`
+  toOmit2 <- setdiff(c(id, blocks$subunit), names(dat))
+  nToOmit2 <- length(toOmit2)
```
Comment (franikowsp) [inline R/checkDesign.r:137]: Nit: Maybe, there is another option for naming both `toOmit` objects, now? 🤔 E.g., (my naive suggestion would be) `toOmitDat` and `toOmitBlocks` (and `nToOmitDat` and `nToOmitBlocks`) or, maybe, just `toOmitDat` and `toOmit`? That might make it easier to identify them.

File: tests/testthat/test-checkDesign.R [hunk 2]
```diff
@@ -180,6 +180,20 @@ test_that_cli("throws warning when more variables in dataset available than in b
...
+    checkDesignTest(dat = within(prepDat, hisei <- NULL))
+  )
+
+  # Manipulation: add two other variables (hisei is available per default)
+  expect_snapshot(
+    checkDesignTest(dat = prepDat[,-c(2,7)])
+  )
+})
+
+
```
Comment (franikowsp) [inline tests/testthat/test-checkDesign.R:196]: LGTM, too!

**Review by franikowsp — APPROVED** [review]
This looks like a very good fix to me; the code and the tests were easy to understand. (There is only one potential, tiny nit concerning the naming of the `toOmit` object remaining.)

File: R/checkDesign.r (hunk 1 above)
Comment (sachseka) [reply R/checkDesign.r]: implemented!

**Pull request merged**